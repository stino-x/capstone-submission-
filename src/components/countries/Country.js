import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import './country.css';
import { getCountry } from '../../redux/Countries/Countries';

const Country = () => {
  const { id } = useParams();
  const country = useSelector(getCountry(id));
  let contentToDisplay = '';
  if (!country) {
    contentToDisplay = <div className="country-status">Loading...</div>;
  } else {
    contentToDisplay = (
      <div className="country-container">
        <div className="country-first">
          <div className="country-flag-container">
            <img
              className="country-flag"
              src={country?.flag}
              alt={country?.name}
            />
          </div>
          <div className="country-data">
            <h3 className="country-title">{country?.name}</h3>
            <p className="country-population">{country?.population}</p>
          </div>
        </div>
        <div className="country-details">
          <h4 className="country-details-title">Country Details</h4>
          <ul className="country-details-list">
            <li className="country-details-item">
              <div className="country-details-item-title">
                Latitude, Longitude
              </div>
              <div className="country-details-item-content">
                {`(${country.latlng[0]}, ${country?.latlng[1]})`}
              </div>
            </li>
            <li className="country-details-item">
              <div className="country-details-item-title">Total Area</div>
              <div className="country-details-item-content">
                {country?.area}
              </div>
            </li>
            <li className="country-details-item">
              <div className="country-details-item-title">Population</div>
              <div className="country-details-item-content">
                {country?.population}
              </div>
            </li>
            <li className="country-details-item">
              <div className="country-details-item-title">Capital</div>
              <div className="country-details-item-content">
                {country?.capital[0]}
              </div>
            </li>
            <li className="country-details-item">
              <div className="country-details-item-title">Region</div>
              <div className="country-details-item-content">
                {country.region}
              </div>
            </li>
            <li className="country-details-item">
              <div className="country-details-item-title">Time Zone</div>
              <div className="country-details-item-content">
                {country?.timezones[0]}
              </div>
            </li>
          </ul>
        </div>
      </div>
    );
  }

  return <div className="country">{contentToDisplay}</div>;
};

export default Country;
