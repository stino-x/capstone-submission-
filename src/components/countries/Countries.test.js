import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Country from './Country';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: '1',
  }),
}));

describe('Country Component', () => {
  let store;
  const mockStore = configureStore([]);

  beforeEach(() => {
    store = mockStore({
      countries: {
        countries: [
          {
            id: 1,
            name: 'Country 1',
            flag: 'https://example.com/flag.png',
            population: 1000000,
            latlng: [12.34, 56.78],
            area: 12345,
            capital: 'Capital City',
            region: 'Region Name',
            timezones: ['Timezone 1'],
          },
        ],
        status: 'succeeded',
      },
    });
  });

  it('should render loading initially', async () => {
    const tree = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/stocks/TEST']}>
          <Country />
        </MemoryRouter>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
