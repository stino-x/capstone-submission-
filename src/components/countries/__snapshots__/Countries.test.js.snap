// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Component should render loading initially 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="country"
      >
        <div
          class="country-container"
        >
          <div
            class="country-first"
          >
            <div
              class="country-flag-container"
            >
              <img
                alt="Country 1"
                class="country-flag"
                src="https://example.com/flag.png"
              />
            </div>
            <div
              class="country-data"
            >
              <h3
                class="country-title"
              >
                Country 1
              </h3>
              <p
                class="country-population"
              >
                1000000
              </p>
            </div>
          </div>
          <div
            class="country-details"
          >
            <h4
              class="country-details-title"
            >
              Country Details
            </h4>
            <ul
              class="country-details-list"
            >
              <li
                class="country-details-item"
              >
                <div
                  class="country-details-item-title"
                >
                  Latitude, Longitude
                </div>
                <div
                  class="country-details-item-content"
                >
                  (12.34, 56.78)
                </div>
              </li>
              <li
                class="country-details-item"
              >
                <div
                  class="country-details-item-title"
                >
                  Total Area
                </div>
                <div
                  class="country-details-item-content"
                >
                  12345
                </div>
              </li>
              <li
                class="country-details-item"
              >
                <div
                  class="country-details-item-title"
                >
                  Population
                </div>
                <div
                  class="country-details-item-content"
                >
                  1000000
                </div>
              </li>
              <li
                class="country-details-item"
              >
                <div
                  class="country-details-item-title"
                >
                  Capital
                </div>
                <div
                  class="country-details-item-content"
                >
                  C
                </div>
              </li>
              <li
                class="country-details-item"
              >
                <div
                  class="country-details-item-title"
                >
                  Region
                </div>
                <div
                  class="country-details-item-content"
                >
                  Region Name
                </div>
              </li>
              <li
                class="country-details-item"
              >
                <div
                  class="country-details-item-title"
                >
                  Time Zone
                </div>
                <div
                  class="country-details-item-content"
                >
                  Timezone 1
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="country"
    >
      <div
        class="country-container"
      >
        <div
          class="country-first"
        >
          <div
            class="country-flag-container"
          >
            <img
              alt="Country 1"
              class="country-flag"
              src="https://example.com/flag.png"
            />
          </div>
          <div
            class="country-data"
          >
            <h3
              class="country-title"
            >
              Country 1
            </h3>
            <p
              class="country-population"
            >
              1000000
            </p>
          </div>
        </div>
        <div
          class="country-details"
        >
          <h4
            class="country-details-title"
          >
            Country Details
          </h4>
          <ul
            class="country-details-list"
          >
            <li
              class="country-details-item"
            >
              <div
                class="country-details-item-title"
              >
                Latitude, Longitude
              </div>
              <div
                class="country-details-item-content"
              >
                (12.34, 56.78)
              </div>
            </li>
            <li
              class="country-details-item"
            >
              <div
                class="country-details-item-title"
              >
                Total Area
              </div>
              <div
                class="country-details-item-content"
              >
                12345
              </div>
            </li>
            <li
              class="country-details-item"
            >
              <div
                class="country-details-item-title"
              >
                Population
              </div>
              <div
                class="country-details-item-content"
              >
                1000000
              </div>
            </li>
            <li
              class="country-details-item"
            >
              <div
                class="country-details-item-title"
              >
                Capital
              </div>
              <div
                class="country-details-item-content"
              >
                C
              </div>
            </li>
            <li
              class="country-details-item"
            >
              <div
                class="country-details-item-title"
              >
                Region
              </div>
              <div
                class="country-details-item-content"
              >
                Region Name
              </div>
            </li>
            <li
              class="country-details-item"
            >
              <div
                class="country-details-item-title"
              >
                Time Zone
              </div>
              <div
                class="country-details-item-content"
              >
                Timezone 1
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
